proxy/http/HttpSessionManager.cc

```diff
@@ -446,6 +466,10 @@ HttpSessionManager::release_session(Http1ServerSession *to_release)
   MUTEX_TRY_LOCK(lock, pool->mutex, ethread);
   if (lock.is_locked()) {
```

Comment:
Hmmm, so the try-lock fails, and we simply just close the session (without this new code)? Why can't we just re-schedule the release event to try again 10ms (or whatever, existing config) later? This wouldn't affect the client I assume (since the txn is done).

Reply:
I did try the reschedule a few years back.  It didn't help things because our global pool is already heavily contended.  Adding retries just adds to the contention.  Even for this PR, I tried a version that attempted to put the sessions back into the global pool on the next use even if we had temporarily put them in the thread pool.  Just that change increased number of connections to origin and reduced connection reuse.  I can give the reschedule another attempt, but probably not until January.  I'm heads down in other issues for the rest of the quarter.